sys/kern/devfs.c

```diff
@@ -74,12 +86,41 @@ static int devfs_add_entry(devfs_node_t *parent, const char *name,
 
   devfs_node_t *dn = kmalloc(M_DEVFS, sizeof(devfs_node_t), M_ZERO);
   dn->dn_ino = ++devfs.next_ino;
+  dn->dn_uid = 0;
+  dn->dn_gid = 0;
+  dn->dn_nlinks = 1;
+  dn->dn_parent = parent;
   dn->dn_name = kstrndup(M_STR, name, DEVFS_NAME_MAX);
   TAILQ_INSERT_TAIL(&parent->dn_children, dn, dn_link);
+  parent->dn_nlinks++;
   *dnp = dn;
   return 0;
 }
 
+static int devfs_vop_getattr(vnode_t *v, vattr_t *va) {
+  devfs_node_t *dn = vn2dn(v);
+  memset(va, 0, sizeof(vattr_t));
+  va->va_mode = dn->dn_mode;
+  va->va_uid = dn->dn_uid;
+  va->va_gid = dn->dn_gid;
+  va->va_nlink = dn->dn_nlinks;
+  va->va_ino = dn->dn_ino;
+  /* XXX we have only devices which should have size 0 */
+  va->va_size = 0;
+  return 0;
+}
+
+void devfs_add_default_vops(vnodeops_t *vops) {
+  if (vops->v_open == NULL)
+    vops->v_open = vnode_open_generic;
+
+  if (vops->v_access == NULL)
+    vops->v_access = vnode_access_generic;
+
+  if (vops->v_getattr == NULL)
+    vops->v_getattr = devfs_vop_getattr;
+}
```

Comment:
Could you make it static or even static inline? I don't see that function outside of devfs.c.

Reply:
Done.